Where you have used the hypothesis testing
I have used the hypothesis testing to validate the distribution of two dataset are same or not 
My Null hypothesis are both the dataset (old and new during the retraining) has same distribution
My alternative distribution was both the dataset has different distribution

In [ ]:
from scipy.

In [5]:
import os
os.getcwd()
os.chdir("c:\\Users\\prash\\Downloads\\Housing_mlops")

In [30]:
os.listdir('housing/artifact/data_ingestion/2024-05-08-22-27-12/ingested_data/train')

['housing.csv']

In [31]:
path="housing/artifact/data_ingestion/2024-05-08-22-27-12/ingested_data/train/housing.csv"

In [34]:
import pandas as pd
df=pd.read_csv(path)

In [35]:
df.shape

(16512, 10)

In [36]:
### Target columns
df['median_house_value']

0         72100.0
1        279600.0
2         82700.0
3        112500.0
4        238300.0
           ...   
16507    268500.0
16508     90400.0
16509    140400.0
16510    258100.0
16511     62700.0
Name: median_house_value, Length: 16512, dtype: float64

In [40]:
x,y=df.drop(columns=['median_house_value']),df[['median_house_value']]

In [38]:
x

longitude  latitude  ...  median_income  ocean_proximity
0        -121.46     38.52  ...         2.1736           INLAND
1        -117.23     33.09  ...         6.3373       NEAR OCEAN
2        -119.04     35.37  ...         2.8750           INLAND
3        -117.13     32.75  ...         2.2264       NEAR OCEAN
4        -118.70     34.28  ...         4.4964        <1H OCEAN
...          ...       ...  ...            ...              ...
16507    -117.07     33.03  ...         5.0900        <1H OCEAN
16508    -121.42     38.51  ...         2.8139           INLAND
16509    -122.72     38.44  ...         3.1797        <1H OCEAN
16510    -122.70     38.31  ...         4.1964        <1H OCEAN
16511    -122.14     39.97  ...         3.1319           INLAND

[16512 rows x 9 columns]

In [46]:
x.shape,y.shape

((16512, 9), (16512, 1))

In [41]:
y

median_house_value
0                 72100.0
1                279600.0
2                 82700.0
3                112500.0
4                238300.0
...                   ...
16507            268500.0
16508             90400.0
16509            140400.0
16510            258100.0
16511             62700.0

[16512 rows x 1 columns]

In [42]:
## check the null values
x.isna().sum()

longitude               0
latitude                0
housing_median_age      0
total_rooms             0
total_bedrooms        158
population              0
households              0
median_income           0
ocean_proximity         0
dtype: int64

158 Null values are available in the Total bedrooms 

In [45]:
x['total_bedrooms'].unique()

array([ 797.,  855.,  310., ..., 1072., 1459., 1661.])

In [50]:
### impute the Null values of all the numerical variables
from sklearn.impute import SimpleImputer

In [62]:
simple_imputer=SimpleImputer(strategy="median")

In [63]:
Numerical_columns=x.drop(columns=['ocean_proximity'])

In [58]:
Numerical_columns

longitude  latitude  ...  households  median_income
0        -121.46     38.52  ...       706.0         2.1736
1        -117.23     33.09  ...       768.0         6.3373
2        -119.04     35.37  ...       300.0         2.8750
3        -117.13     32.75  ...       483.0         2.2264
4        -118.70     34.28  ...       580.0         4.4964
...          ...       ...  ...         ...            ...
16507    -117.07     33.03  ...      1001.0         5.0900
16508    -121.42     38.51  ...      1418.0         2.8139
16509    -122.72     38.44  ...       172.0         3.1797
16510    -122.70     38.31  ...       501.0         4.1964
16511    -122.14     39.97  ...       197.0         3.1319

[16512 rows x 8 columns]

In [64]:
simple_imputer.fit(Numerical_columns)

SimpleImputer(strategy='median')

In [65]:
simple_imputer.transform(Numerical_columns)

array([[-1.2146e+02,  3.8520e+01,  2.9000e+01, ...,  2.2370e+03,
         7.0600e+02,  2.1736e+00],
       [-1.1723e+02,  3.3090e+01,  7.0000e+00, ...,  2.0150e+03,
         7.6800e+02,  6.3373e+00],
       [-1.1904e+02,  3.5370e+01,  4.4000e+01, ...,  6.6700e+02,
         3.0000e+02,  2.8750e+00],
       ...,
       [-1.2272e+02,  3.8440e+01,  4.8000e+01, ...,  4.5800e+02,
         1.7200e+02,  3.1797e+00],
       [-1.2270e+02,  3.8310e+01,  1.4000e+01, ...,  1.2080e+03,
         5.0100e+02,  4.1964e+00],
       [-1.2214e+02,  3.9970e+01,  2.7000e+01, ...,  6.2500e+02,
         1.9700e+02,  3.1319e+00]])

In [66]:
simple_imputer.statistics_

array([-118.51   ,   34.26   ,   29.     , 2119.     ,  433.     ,
       1164.     ,  408.     ,    3.54155])

In [67]:
simple_imputer.feature_names_in_

array(['longitude', 'latitude', 'housing_median_age', 'total_rooms',
       'total_bedrooms', 'population', 'households', 'median_income'],
      dtype=object)